huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go

```diff
@@ -891,6 +930,94 @@ master_nodes {
 }`, testAccMrsMapReduceClusterConfig_base(rName), rName, pwd, pwd, nodeNum1)
 }
 
+func testAccMrsClusterConfig_externalDataSources(rName, pwd string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_availability_zones" "test" {}
+
+resource "huaweicloud_rds_instance" "test" {
+  name                = "%[2]s"
+  flavor              = "rds.mysql.sld4.large.ha"
+  security_group_id   = huaweicloud_networking_secgroup.test.id
+  subnet_id           = huaweicloud_vpc_subnet.test.id
+  vpc_id              = huaweicloud_vpc.test.id
+  fixed_ip            = "192.168.0.58"
+  ha_replication_mode = "semisync"
+
+  availability_zone = [
+    data.huaweicloud_availability_zones.test.names[0],
+    data.huaweicloud_availability_zones.test.names[3],
+  ]
+
+  db {
+    password = "%[3]s"
+    type     = "MySQL"
+    version  = "5.7"
+    port     = 3306
+  }
+
+  volume {
+    type = "LOCALSSD"
+    size = 40
+  }
+}
+
+resource "huaweicloud_rds_mysql_database" "test" {
+  instance_id   = huaweicloud_rds_instance.test.id
+  name          = "%[2]s"
+  character_set = "utf8"
+}
+
+resource "huaweicloud_mapreduce_connection" "test" {
+  type        = "RDS_MYSQL"
+  name        = "%s"
+  instance_id = huaweicloud_rds_instance.test.id
+  database    = huaweicloud_rds_mysql_database.test.id
+  username    = "root"
+  password    = "%[3]s"
+}
+
+resource "huaweicloud_mapreduce_cluster" "test" {
+  availability_zone  = data.huaweicloud_availability_zones.test.names[0]
+  name               = "%[2]s"
+  type               = "ANALYSIS"
+  version            = "MRS 1.9.2"
+  safe_mode          = false
+  manager_admin_pass = "%[3]s"
+  node_admin_pass    = "%[3]s"
+  vpc_id             = "60f6bd6b-3b21-4f07-b5e0-f2f876410902" //huaweicloud_vpc.test.id
+  subnet_id          = "828907cc-40c9-42fe-8206-ecc1bdd30060" //huaweicloud_vpc_subnet.test.id
+  component_list     = ["Hadoop", "Hive", "Tez"]
+
+  master_nodes {
+    flavor            = "c3ne.xlarge.4.linux.bigdata"
+    node_number       = 2
+    root_volume_type  = "SAS"
+    root_volume_size  = 600
+    data_volume_type  = "SAS"
+    data_volume_size  = 600
+    data_volume_count = 1
+  }
+  analysis_core_nodes {
+    flavor            = "c3ne.xlarge.4.linux.bigdata"
+    node_number       = 3
+    root_volume_type  = "SAS"
+    root_volume_size  = 600
+    data_volume_type  = "SAS"
+    data_volume_size  = 600
+    data_volume_count = 1
+  }
+
+  external_datasources {
+    component_name = "Hive"
+    role_type      = "hive_metastore"
+    source_type    = "RDS_MYSQL"
+    connector_id   = huaweicloud_mapreduce_connection.test.id
```

Comment:
replace with `HW_MAPREDUCE_EXTERNAL_DATASOURCES` ?

Reply:
huaweicloud_mapreduce_connection is declared above, which is not ready currently.